src/components/PostDetails.tsx

```diff
@@ -1,106 +1,94 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import * as commentService from '../api/comments';
 import { Loader } from './Loader';
+import { CommentList } from './CommentList';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+
+type Props = {
+  selectedPost: Post;
+};
+
+export const PostDetails: React.FC<Props> = ({
+  selectedPost: selectedPost,
+}) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isCommentsLoading, setIsCommentsLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [isCommentFormShown, setIsCommentFormShown] = useState(false);
+
+  useEffect(() => {
+    setIsCommentsLoading(true);
+    commentService
+      .getComments(selectedPost.id)
+      .then(setComments)
+      .catch(() => setHasError(true))
+      .finally(() => setIsCommentsLoading(false));
+  }, [selectedPost]);
+
+  function deleteComment(commId: number) {
+    setComments(currComments => currComments.filter(com => commId !== com.id));
+
+    return commentService.deleteComment(commId).catch(error => {
+      setComments(comments);
+      setHasError(true);
+      throw error;
```

Comment:
Throwing an error inside the `catch` block will not be caught by any surrounding code, as this is an asynchronous operation. Consider logging the error or handling it in a way that provides feedback to the user.